server/src/main/java/org/opensearch/bootstrap/Security.java

```diff
@@ -231,33 +235,45 @@ static Policy readPolicy(URL policyFile, Map<String, URL> codebases) {
         try {
             List<String> propertiesSet = new ArrayList<>();
             try {
+                final Map<Map.Entry<String, URL>, String> jarsWithPossibleClassifiers = new HashMap<>();
```

Comment:
I have to admit a map with a map entry as a key took me a few minutes to figure out...

Reply:
Sorry about that, I was following the least disruptive path (the code is very much isolated in here), let me know if you would like me to make it more readable